modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala

```diff
@@ -84,12 +82,16 @@ object AvroSpanExporter {
         .drain
 
     for {
-      avroSchema <- Resource.liftF(AvroInstances.completedSpanSchema[F])
-      address <- Resource.liftF(Sync[F].delay(new InetSocketAddress(host, port)))
-      queue <- Resource.liftF(InspectableQueue.bounded[F, Batch[G]](1))
-      semaphore <- Resource.liftF(Semaphore[F](Long.MaxValue))
-      socketGroup <- UDPSocketGroup(blocker)
-      socket <- socketGroup.open()
+      avroSchema <- Resource.eval(AvroInstances.completedSpanSchema[F])
+      host <- Resource.eval(IpAddress.fromString(host).liftTo[F](new IllegalArgumentException(s"invalid host $host")))
+      port <- Resource.eval(Port.fromInt(port).liftTo[F](new IllegalArgumentException(s"invalid port $port")))
+      address = SocketAddress(host, port)
+      queue <- Resource.eval(
+        Queue.bounded[F, Batch[G]](1)
+      ) //TODO: replace with Ref of Option or Queue of Option and noneTerminate?
+      semaphore <- Resource.eval(Semaphore[F](Long.MaxValue))
+      socketGroup <- Network[F].datagramSocketGroup() //TODO: maybe we should use the global datagramSocketGroup
```

Comment:
https://gitter.im/functional-streams-for-scala/fs2?at=603fdc6ad74bbe49e0cf8fe5

There's no reason to use manually managed socket groups